_plugins/rosindex_generator.rb

```diff
@@ -655,8 +656,14 @@ def generate_sorted_paginated(site, elements_sorted, default_sort_key, n_element
 
       elements_sorted.each do |sort_key, elements|
         # Get a subset of the elements
-        elements_sliced = elements.slice(p_start, elements_per_page)
-
+        unless sort_key == 'name' then
+          elements_sliced = Hash.new
+          elements.each do |distro, items|
+            elements_sliced[distro] = items.slice(p_start, elements_per_page)
+          end
+        else
+          elements_sliced = elements.slice(p_start, elements_per_page)
+        end
```

Comment:
@apojomovsky shouldn't `elements_sliced` always be a hash map of _things_ per distro?